indexing-hadoop/src/test/java/io/druid/indexer/path/GranularityPathSpecTest.java

```diff
@@ -85,6 +114,70 @@ public void testSerdeNoInputFormat() throws Exception
     testSerde("/test/path", "*.test", "pat_pat", Granularity.SECOND, null);
   }
 
+  @Test
+  public void testAddInputPath() throws Exception
+  {
+    Iterable<FileStatus> testIterable = new Iterable<FileStatus>() {
+      private int j;
+
+      @Override
+      public Iterator<FileStatus> iterator() {
+        if (++j % 2 == 0) {
+          return new FSSpideringIterator(new LocalFileSystem(), new FileStatus[]{});
+        }
+
+        return new Iterator<FileStatus>() {
+          private int i;
+
+          @Override
+          public boolean hasNext() {
+            return ++i <= 3;
+          }
+
+          @Override
+          public FileStatus next() {
+            return new FileStatus(0L, false, 0, 0L, 0L, new Path(String.format("file:///test/path/%d/%d.gz", j, i)));
+          }
+
+          @Override
+          public void remove() { }
+        };
+      }
+    };
+
+    UserGroupInformation.setLoginUser(UserGroupInformation.createUserForTesting("test", new String[]{"testGroup"}));
+    granularityPathSpec.setDataGranularity(Granularity.HOUR);
+    granularityPathSpec.setInputPath("/test/path");
+    granularityPathSpec.setFilePattern("file:/+test/path/\\d+/\\d+.gz");
+    granularityPathSpec.setInputFormat(CombineTextInputFormat.class);
+
+    HadoopDruidIndexerConfig mockIndexerConfig = EasyMock.createNiceMock(HadoopDruidIndexerConfig.class);
+    Capture<String> pathCapture = Capture.newInstance(CaptureType.ALL);
+    Job job = Job.getInstance();
+
+    PowerMock.mockStatic(FSSpideringIterator.class);
```

Comment:
is it not possible to setup the appropriate files in some temporary space on disk that reproduces the bug instead of adding PowerMock ?